Plugin fails if the 'target' dir does not exist

In case of Maven projects with `<packaging>pom</packaging>` and not any other steps then creating a container based on a Containerfile, there is no target directory created automatically which makes the build fail.

If `mkdir target/` is executed then the build happily continues.

Maybe have the plugin create the target dir when executing the build?

Here the `pom.xml` I used:

<details>

```xml
<project xmlns = "http://maven.apache.org/POM/4.0.0"
   xmlns:xsi = "http://www.w3.org/2001/XMLSchema-instance"
   xsi:schemaLocation = "http://maven.apache.org/POM/4.0.0
   http://maven.apache.org/xsd/maven-4.0.0.xsd">
   <modelVersion>4.0.0</modelVersion>

   <groupId>com.roylenferink.test-projects</groupId>
   <artifactId>multi-stage-container-builds</artifactId>
   <version>1.0.0-SNAPSHOT</version>
   <packaging>pom</packaging>

   <properties>
     <baseImage.name>docker.io/centos</baseImage.name>
     <baseImage.version>centos7.9.2009</baseImage.version>
     <imageName>rlenferink/test-image</imageName>
   </properties>

   <build>
     <plugins>
       <plugin>
         <groupId>nl.lexemmens</groupId>
         <artifactId>podman-maven-plugin</artifactId>
         <version>1.8.0-SNAPSHOT</version>
         <configuration>
           <skipAuth>true</skipAuth>
           <pushRegistry>localhost</pushRegistry>
           <deleteLocalImageAfterPush>true</deleteLocalImageAfterPush>
           <images>
             <image>
               <name>${imageName}</name>
               <customImageNameForMultiStageContainerfile>true</customImageNameForMultiStageContainerfile>
               <stages>
                 <stage>
                   <name>devel</name>
                   <imageName>${imageName}-devel</imageName>
                 </stage>
                 <stage>
                   <name>run</name>
                   <imageName>${imageName}-run</imageName>
                 </stage>
               </stages>
               <build>
                 <tagWithMavenProjectVersion>true</tagWithMavenProjectVersion>
                 <createLatestTag>true</createLatestTag>
               </build>
             </image>
           </images>
         </configuration>
         <executions>
           <execution>
             <id>clean-image</id>
             <phase>clean</phase>
             <goals>
               <goal>clean</goal>
             </goals>
           </execution>
           <execution>
             <id>build-image</id>
             <phase>package</phase>
             <goals>
               <goal>build</goal>
             </goals>
           </execution>
         </executions>
       </plugin>
     </plugins>
   </build>

</project>
```

<summary>pom.xml</summary>
</details


Used `Containerfile`:

```bash
FROM ${baseImage.name}:${baseImage.version} AS devel

FROM ${baseImage.name}:${baseImage.version} AS run
```

Error message:
```
[INFO] --- podman-maven-plugin:1.8.0-SNAPSHOT:build (build-image) @ multi-stage-container-builds ---
[INFO] Registry authentication is skipped.
[INFO] Filtering Containerfile...
[ERROR] Failed to filter Containerfile! /home/rlenferink/workspace/lexemmens/test-project/target/Containerfile (No such file or directory)
org.apache.maven.shared.filtering.MavenFilteringException: /home/rlenferink/workspace/lexemmens/test-project/target/Containerfile (No such file or directory)
    at org.apache.maven.shared.filtering.DefaultMavenFileFilter.copyFile (DefaultMavenFileFilter.java:128)
    at org.apache.maven.shared.filtering.DefaultMavenFileFilter.copyFile (DefaultMavenFileFilter.java:97)
    at org.apache.maven.shared.filtering.DefaultMavenFileFilter.copyFile (DefaultMavenFileFilter.java:87)
    at nl.lexemmens.podman.service.ContainerfileDecorator.filterContainerfile (ContainerfileDecorator.java:94)
    at nl.lexemmens.podman.service.ContainerfileDecorator.decorateContainerfile (ContainerfileDecorator.java:80)
    at nl.lexemmens.podman.BuildMojo.decorateContainerfile (BuildMojo.java:70)
    at nl.lexemmens.podman.BuildMojo.executeInternal (BuildMojo.java:60)
    at nl.lexemmens.podman.AbstractPodmanMojo.execute (AbstractPodmanMojo.java:132)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: java.io.FileNotFoundException: /home/rlenferink/workspace/lexemmens/test-project/target/Containerfile (No such file or directory)
    at java.io.FileOutputStream.open0 (Native Method)
    at java.io.FileOutputStream.open (FileOutputStream.java:270)
    at java.io.FileOutputStream.<init> (FileOutputStream.java:213)
    at java.io.FileOutputStream.<init> (FileOutputStream.java:162)
    at org.apache.maven.shared.filtering.DefaultMavenFileFilter.getFileWriter (DefaultMavenFileFilter.java:173)
    at org.apache.maven.shared.filtering.DefaultMavenFileFilter.filterFile (DefaultMavenFileFilter.java:144)
    at org.apache.maven.shared.filtering.DefaultMavenFileFilter.copyFile (DefaultMavenFileFilter.java:113)
    at org.apache.maven.shared.filtering.DefaultMavenFileFilter.copyFile (DefaultMavenFileFilter.java:97)
    at org.apache.maven.shared.filtering.DefaultMavenFileFilter.copyFile (DefaultMavenFileFilter.java:87)
    at nl.lexemmens.podman.service.ContainerfileDecorator.filterContainerfile (ContainerfileDecorator.java:94)
    at nl.lexemmens.podman.service.ContainerfileDecorator.decorateContainerfile (ContainerfileDecorator.java:80)
    at nl.lexemmens.podman.BuildMojo.decorateContainerfile (BuildMojo.java:70)
    at nl.lexemmens.podman.BuildMojo.executeInternal (BuildMojo.java:60)
    at nl.lexemmens.podman.AbstractPodmanMojo.execute (AbstractPodmanMojo.java:132)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.271 s
[INFO] Finished at: 2021-09-25T11:20:37+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal nl.lexemmens:podman-maven-plugin:1.8.0-SNAPSHOT:build (build-image) on project multi-stage-container-builds: Failed to filter Containerfile! /home/rlenferink/workspace/lexemmens/test-project/target/Containerfile (No such file or directory) -> [Help 1]
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,6 @@
 ## Changelog
 ### UNRELEASED CHANGES
-* None
+* ([#49](https://github.com/lexemmens/podman-maven-plugin/issues/49)) - Using properties in the FROM now succeeds for multistage builds.
 
 ### 1.7.1 (16-08-2021)
 #### Bugs

--- a/src/main/java/nl/lexemmens/podman/config/image/AbstractImageBuildConfiguration.java
+++ b/src/main/java/nl/lexemmens/podman/config/image/AbstractImageBuildConfiguration.java
@@ -26,10 +26,12 @@ import static nl.lexemmens.podman.enumeration.ContainerFormat.OCI;
 public abstract class AbstractImageBuildConfiguration {
 
     /**
-     * This is the regular expression to be used to determine a multistage Containerfiles. For now we only support
+     * This is the regular expression to be used to determine a multistage Containerfile. For now we only support
      * named stages.
      */
-    protected static final Pattern MULTISTAGE_CONTAINERFILE_REGEX = Pattern.compile("(FROM\\s[a-zA-Z0-9./:_\\-]{0,255}\\s)([ASas]{2}\\s)([a-zA-Z0-9./:_\\-]{1,128})");
+    protected static final Pattern MULTISTAGE_CONTAINERFILE_REGEX = Pattern.compile(
+        "(FROM\\s[a-zA-Z0-9./:_\\-${}]{0,255}\\s)([ASas]{2}\\s)([a-zA-Z0-9./:_\\-]{1,128})"
+    );
 
     /**
      * The default name of the Containerfile to build.

